api/src/controllers/document-controller.js

```diff
@@ -54,6 +54,8 @@ exports.protectedPost = async function(args, res, next) { // Confirm user has co
           args.swagger.params.url.value,
           readRoles
         );
+
+        await queryUtils.recordAction('POST', JSON.stringify(docResponse), args.swagger.params.auth_payload, docResponse.key);
```

Comment:
Its not super obvious, but this `docResponse` will never have a `key`.

The `key` is the S3 key, so it only gets created if you create a document via `createS3Document()`

Reply:
Just noticed: I think that parameter wants the object _id, so would want to pass `docResponse._id`.  I think this applies to all of the recordActions where `docResponse.key` is being set.

Reply:
Sounds good. I'll update those now.